src/Calendar.js

```diff
@@ -40,6 +39,19 @@ const Calendar = ({
     cycleCount: 1,
     activeDate: null,
   });
+  const [firstSelectedDate, setFirstSelectedDate] = useState(null);
+  const [secondSelectedDate, setSecondSelectedDate] = useState(null);
+  const [currentlyDragging, setCurrentlyDragging] = useState(false);
```

Comment:
I think this should be rather a computed state than a normal state value.

Reply:
Do you mean computing them based on initial selectedDayRange values? In this case, I agree with you and I'll fix them

Reply:
That one as well, but I mean `currentlyDragging` shouldn't be a state value. It gotta be a computed variable based on `firstSelectedDate` and `secondSelectedDate` and **the state of the mouse down/up**. It looks like **it won't trigger a necessary render by the change of its value**. We will discuss this on the "making it fast" phase😊

Reply:
Yeah!!! you're right. I haven't noticed it (yet😂)
we can easily use "useRef" in order to track which end of selectedRange is getting dragged.